torch/_inductor/scheduler.py

```diff
@@ -2742,64 +2793,108 @@ def log_fusion(ms_fused: float, ms1: float, ms2: float) -> None:
                 if triton_choices > config.max_epilogue_benchmarked_choices:
                     break
 
-                # TODO - parallel compile triton templates
-                # TODO - should prune/skip choices that are not within certain % of best choice
                 with multi_node.swap_as_triton_caller(choice):
-                    ms_fused, path = self.benchmark_fused_nodes(node_list_fused)
+                    future_choices.append((choice, *compile_kernel(node_list_fused)))
 
-                    if ms_fused < min_ms_fused:
-                        min_ms_fused = ms_fused
-                        ms_fused_choice = choice
+            if len(future_choices) == 0:
+                return False
 
-            log_fusion(min_ms_fused, ms1, ms2)
+            def benchmark_when_ready() -> bool:
+                min_ms_fused = float("inf")
+                ms_fused_choice = None
+
+                # Benchmark each choice after compilation completes
+                for choice, future, mod_fused in future_choices:
+                    if future is not None:
+                        future.result()
```

Comment:
Any concern about the ongoing parallel compiles messing up timings?

Reply:
We do the same thing for base mm compile + benchmarking. so not overly concerned. but open to suggestions if theres anything else we should do to handle this